test-records.js

```diff
@@ -77,26 +93,76 @@ validator('companies', cdb_schema, (json) => {
         if (json[prop]) {
             if (json['request-language']) {
                 if (!templates[json['request-language']].includes(json[prop]))
-                    fail(
-                        `Record specifies '${prop}' of '${json[prop]}' but that isn't available for 'request-language' of '${json['request-language']}'.`
-                    );
+                    errors.push({
+                        msg: `Record specifies '${prop}' of '${json[prop]}' but that isn't available for 'request-language' of '${json['request-language']}'.`,
+                    });
             } else {
                 if (!templates['en'].includes(json[prop]))
-                    fail(
-                        `Record specifies '${prop}' of '${json[prop]}' but that isn't available in English.`,
-                        'See: https://github.com/datenanfragen/data/issues/1120'
-                    );
+                    errors.push({
+                        msg: `Record specifies '${prop}' of '${json[prop]}' but that isn't available in English.`,
+                        ref: 'https://github.com/datenanfragen/data/issues/1120',
+                    });
             }
         }
     }
 
     // A `quality` of `tested` may only be set if `required-elements` are specified (#811).
     if (json['quality'] === 'tested') {
         if (!json['required-elements'])
-            fail(
-                "Record has `quality` of `tested` but doesn't specify `required-elements`.",
-                'See: https://github.com/datenanfragen/data/issues/811'
+            errors.push({
+                msg: "Record has `quality` of `tested` but doesn't specify `required-elements`.",
+                ref: 'https://github.com/datenanfragen/data/issues/811',
+            });
+    }
+    // whitespace check
+    Object.keys(json).forEach((key) => {
+        if (typeof json[key] === 'string' && json[key] !== json[key].trim())
+            errors.push(`Seems like \`${key}\` isn't trimmed, i.e. it contains leading or trailing whitespace.`);
+    });
+
+    // address formatting
+    const address_lines = json['address'].split('\n');
+    if (address_lines.length < 2) errors.push('`address` is not formatted with newlines (\\n).');
+    for (line of address_lines) {
```

Comment:
```suggestion
    for (const line of address_lines) {
```

Reply:
Oh—and that could of course once again be a `address_lines.some(...)`. :D

Reply:
On another day. :D